### PR TITLE
[double-take] Make detectors and notifiers optional

### LINKED ISSUE #1314: double-take please make compreface optional
# Details

**Helm chart name and version:**
double-take-1.1.0

**Container name and tag:**
1.6.0

**What steps did you take and what happened:**
Install double-take and try to use facebox instead of compreface.

These are the errors in the logfile:
```
double-take error: detectors.compreface.url is not of a type(s) string   
double-take error: notify.gotify.url is not of a type(s) string    
```

**Relevant Helm values:**
```
image:
  repository: jakowenko/double-take
  tag: 1.7.0
env:
  TZ: "Europe/Berlin"
config:
  mqtt:
    host: mosquitto
  frigate:
    url: http://frigate:5000
  detectors:
    # compreface:
    #   url: http://compreface:8080
    # deepstack:
    #   url: http://deepstack:5000
    facebox:
      url: http://facebox:8080
```

**What did you expect to happen:**
The helm chart should not add a compreface section if there is no url specified.

**Anything else you would like to add:**
The same issue happens with the gotify stuff,

**Additional Information:**

### PR DESCRIPTION
**Pull request by Jonnobrow**
**Description of the change**

Each detector and notifier (gotify) now contains an `enabled` field
which must be set to true to add that detector to the config, otherwise
it will be omitted.

**Benefits**
- There is no longer a requirement to use the `compreface` detector

**Possible drawbacks**
- Exisiting configurations will need to add the `enabled ` key to the `deepstack` and `facebox` detectors in order for their configuration to be added to the config map. 

**Applicable issues**

- fixes #1314

**Additional information**

**Checklist** 
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Title of the PR starts with chart name (e.g. `[home-assistant]`)
- [X] Variables are documented in the README.md (this can be done with using our helm-docs wrapper `./hack/gen-helm-docs.sh stable <chart>`)

### CODE REVIEW

File: charts/stable/double-take/templates/configmap.yaml [hunk 1]
```diff
@@ -7,6 +7,18 @@ metadata:
...
-{{- with .Values.config }}
-    {{- toYaml . | nindent 4 }}
+{{- range $name, $config := .Values.config }}
+{{- if or (eq $name "detectors") (eq $name "notify") }}
+    {{ $name }}:
+      {{- range $innerName, $innerConfig := $config }}
+        {{- if $innerConfig.enabled }}
+        {{- $newInnerConfig := omit $innerConfig "enabled" }}
+      {{ $innerName }}:
+{{ toYaml $newInnerConfig | indent 8 }}
```
Comment (bjw-s) [inline charts/stable/double-take/templates/configmap.yaml:17]: I would indent this "properly" with the rest of the code and then replace it with
`{{- toYaml $newInnerConfig | nindent 8 }}`. (adjust the 8 as required)

File: charts/stable/double-take/templates/configmap.yaml [hunk 2]
```diff
@@ -7,6 +7,18 @@ metadata:
...
+        {{- end }}
+      {{- end }}
+{{- else }}
+    {{ $name }}:
+{{ toYaml $config | indent 6 }}
```
Comment (bjw-s) [inline charts/stable/double-take/templates/configmap.yaml:22]: I would indent this "properly" with the rest of the code and then replace it with
`{{- toYaml $config | nindent 6 }}`. (adjust the 6 as required)

File: charts/stable/double-take/values.yaml [hunk 3]
```diff
@@ -206,16 +209,22 @@ config:
...
-     #   key:
-     #   # number of seconds before the request times out and is aborted
-     #   timeout: 15
+    deepstack:
+      # -- enable or disable the compreface detector
+      # @default -- see values.yaml
+      enabled: false
+      url:
+      key:
+      # number of seconds before the request times out and is aborted
```
Comment (bjw-s) [inline charts/stable/double-take/values.yaml:218]:
```suggestion
      # -- number of seconds before the request times out and is aborted
```

File: charts/stable/double-take/values.yaml [hunk 4]
```diff
@@ -206,16 +209,22 @@ config:
...
-     # facebox:
-     #   url:
-     #   # number of seconds before the request times out and is aborted
-     #   timeout: 15
+    facebox:
+      # -- enable or disable the compreface detector
+      # @default -- see values.yaml
+      enabled: false
+      url:
+      # number of seconds before the request times out and is aborted
```
Comment (bjw-s) [inline charts/stable/double-take/values.yaml:226]:
```suggestion
      # -- number of seconds before the request times out and is aborted
```

File: charts/stable/double-take/values.yaml [hunk 5]
```diff
@@ -240,6 +249,9 @@ config:
   # notify settings (default: shown below)
   notify:
     gotify:
+      # -- enable gotify
+      # @default -- see values.yaml
```
Comment (bjw-s) [inline charts/stable/double-take/values.yaml:253]:
```suggestion
```

File: charts/stable/double-take/values.yaml [hunk 6]
```diff
@@ -206,16 +209,22 @@ config:
...
+      # number of seconds before the request times out and is aborted
+      timeout: 15
 
-     # facebox:
-     #   url:
-     #   # number of seconds before the request times out and is aborted
-     #   timeout: 15
+    facebox:
+      # -- enable or disable the compreface detector
+      # @default -- see values.yaml
```
Comment (bjw-s) [inline charts/stable/double-take/values.yaml:223]:
```suggestion
```

File: charts/stable/double-take/values.yaml [hunk 7]
```diff
@@ -206,16 +209,22 @@ config:
...
+      key:
+      # number of seconds before the request times out and is aborted
+      timeout: 15
 
-     # facebox:
-     #   url:
-     #   # number of seconds before the request times out and is aborted
-     #   timeout: 15
+    facebox:
+      # -- enable or disable the compreface detector
```
Comment (bjw-s) [inline charts/stable/double-take/values.yaml:222]:
```suggestion
      # -- enable or disable the facebox detector
```

File: charts/stable/double-take/values.yaml [hunk 8]
```diff
@@ -206,16 +209,22 @@ config:
...
       # face_plugins: mask,gender,age
 
-     # deepstack:
-     #   url:
-     #   key:
-     #   # number of seconds before the request times out and is aborted
-     #   timeout: 15
+    deepstack:
+      # -- enable or disable the compreface detector
+      # @default -- see values.yaml
```
Comment (bjw-s) [inline charts/stable/double-take/values.yaml:214]:
```suggestion
```

File: charts/stable/double-take/values.yaml [hunk 9]
```diff
@@ -206,16 +209,22 @@ config:
       # https://github.com/exadel-inc/CompreFace/blob/master/docs/Face-services-and-plugins.md)
       # face_plugins: mask,gender,age
 
-     # deepstack:
-     #   url:
-     #   key:
-     #   # number of seconds before the request times out and is aborted
-     #   timeout: 15
+    deepstack:
+      # -- enable or disable the compreface detector
```
Comment (bjw-s) [inline charts/stable/double-take/values.yaml:213]:
```suggestion
      # -- enable or disable the deepstack detector
```

File: charts/stable/double-take/values.yaml [hunk 10]
```diff
@@ -194,6 +194,9 @@ config:
   # detector settings (default: shown below)
   detectors:
     compreface:
+      # -- enable or disable the compreface detector
+      # @default -- see values.yaml
```
Comment (bjw-s) [inline charts/stable/double-take/values.yaml:198]:
```suggestion
```

**Review by bjw-s — CHANGES_REQUESTED** [review]

**Comment by Jonnobrow** [comment]
Thanks for the review! I have applied the suggested changes. Please let me know if there is anything else I can do 😃

**Review by bjw-s — APPROVED** [review]

**Pull request merged**